amazon-kinesis-client/src/main/java/software/amazon/kinesis/processor/MultiStreamTracker.java

```diff
@@ -0,0 +1,19 @@
+package software.amazon.kinesis.processor;
+
+import software.amazon.kinesis.common.StreamConfig;
+
+import java.util.Map;
+
+/**
+ * Interface for stream trackers. This is useful for KCL Workers that need
+ * to consume data from multiple streams.
+ */
+public interface MultiStreamTracker {
+
+    /**
+     * Returns the map of streams and its associated stream specific config.
+     *
+     * @return List of stream names
+     */
+    Map<String, StreamConfig> streamConfigMap();
```

Comment:
This iface even be used for singleStream setup. anything blocking us using this iface as the default for all applications? we would be just calling this just StreamTracker in that case.

Reply:
The idea is customer who implement this interface will see change in their ddb schema as their lease object will be different now. 

In scheduler, we use this to differentiate between Multistream and single stream in order to initialize lease serializer, lease manager, shard syncer etc.

`private final Either<MultiStreamTracker, StreamConfig> appStreamTracker;`

Also in scheduler, we use the following to track both multistream and single stream eventually.

`private final Map<String, StreamConfig> currentStreamConfigMap;`